crates/rmcp/src/handler/server/tool_name_validation.rs

```diff
@@ -0,0 +1,261 @@
+//! Tool name validation utilities according to SEP: Specify Format for Tool Names
+//!
+//! Tool names SHOULD be between 1 and 128 characters in length (inclusive).
+//! Tool names are case-sensitive.
+//! Allowed characters: uppercase and lowercase ASCII letters (A-Z, a-z), digits
+//! (0-9), underscore (_), dash (-), and dot (.).
+//! Tool names SHOULD NOT contain spaces, commas, or other special characters.
+
+use std::collections::HashSet;
+
+/// Result of tool name validation containing validation status and warnings.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct ToolNameValidationResult {
+    /// Whether the tool name is valid according to the specification
+    pub is_valid: bool,
+    /// Array of warning messages about non-conforming aspects of the tool name
+    pub warnings: Vec<String>,
+}
+
+impl ToolNameValidationResult {
+    /// Create a new validation result
+    pub fn new(is_valid: bool, warnings: Vec<String>) -> Self {
+        Self { is_valid, warnings }
+    }
+}
+
+/// Validates a tool name according to the SEP specification.
+pub fn validate_tool_name(name: &str) -> ToolNameValidationResult {
```

Comment:
these don't need to be `pub` right?

only `validate_and_warn_tool_name`

Reply:
Given that the tool name validation is only for the purpose of logging warnings right now, I think we should keep the other things private